src/migration/1714662136871-SquadTotalUpvotesTrigger.ts

```diff
@@ -0,0 +1,47 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class SquadTotalUpvotesTrigger1714662136871
+  implements MigrationInterface
+{
+  name = 'SquadTotalUpvotesTrigger1714662136871';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    queryRunner.query(`
+        CREATE OR REPLACE FUNCTION update_squad_upvotes_count()
+          RETURNS TRIGGER
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        BEGIN
+          IF NEW.upvotes <> OLD.upvotes THEN
+            UPDATE  source
+            SET     flags = jsonb_set(
+                              flags,
+                              '{totalUpvotes}',
+                              to_jsonb(
+                                GREATEST(
+                                  0,
+                                  COALESCE(CAST(flags->>'totalUpvotes' AS INTEGER), 0) +
+                                  (CASE WHEN NEW.upvotes > OLD.upvotes THEN 1 ELSE -1 END)
+                                )
+                              )
+                            )
+            WHERE   id = NEW."sourceId"
+            AND     type = 'squad';
+          END IF;
+          RETURN NEW;
+        END;
+        $$
+      `);
+    queryRunner.query(
+      `CREATE TRIGGER update_squad_upvotes_count AFTER UPDATE ON "post" FOR EACH ROW EXECUTE PROCEDURE update_squad_upvotes_count()`,
```

Comment:
More of a question @capJavert is there no way we can only attach trigger to set column?

Reply:
Alternative, we listen to the same thing here, wondering if it's "cheaper" to simply just do all with 1 trigger?

Reply:
Hmm I think each set of triggers should listen on the source table:
- totalUpvotes triggers to `user_post` table
- totalPosts triggers `post` table (as it is now)
- totalView triggers on `view` table

Reply:
As to listening to specific table that might not be possible due to https://dailydotdev.slack.com/archives/C069XNYMDGV/p1714669514654279

Reply:
Disregard my comment on combining and column listening :)

Reply:
Yeah, it would have been ideal to listen to each those tables. The only downside is that we have to fetch the source id from the parent entity which is the post (for view and user_post).

Reply:
If we are cool doing for that, I'd be glad to attach the triggers closest to where the action is being recorded.

Reply:
Yeah, lets close this discussion and do it on post as you did, and in regards to slack thread.